### PR TITLE
Adds additional fields to tag schema

### PR DESCRIPTION
**Pull request by mikemurray**
- Adds optional type field for better categorization needs
- Adds optional metafields array for instances where schema may fall short
- Adds isVisible field, useful for hiding a tag from showing in a list
- Adds groups array, allowing for hiding tags by specific user groups

### CODE REVIEW

File: lib/collections/schemas/tags.js [hunk 1]
```diff
@@ -17,6 +18,14 @@ export const Tag = new SimpleSchema({
   slug: {
     type: String
   },
+  type: {
```
Comment (zenweasel) [inline lib/collections/schemas/tags.js]: type seems sort of vague? Just reading this I am not clear how I would use it?

Comment (aaronjudd) [reply lib/collections/schemas/tags.js]: `type` has been (my)/the preferred name for anything that determines functionality, and should be a fairly standard behavior expectation across our schemas. 👍

**Comment by aaronjudd** [comment]
`isVisible` feels like it should actually be a security group related filter, but... I know we've used the same convention elsewhere.

**Comment by jshimko** [comment]
After merging this into marketplace, I now see this error when restarting a shop with an existing database.

reaction

Missed migration or something?

**Comment by mikemurray** [comment]
Probably the `isVisible` field, since it's not marked optional.

**Pull request merged**